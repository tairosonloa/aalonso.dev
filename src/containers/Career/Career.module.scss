@import 'src/styles/utils.scss';
@import 'src/styles/palette.scss';

.career {
  display: flex;
  position: relative;
  flex-direction: column;
  width: 100%;
  max-width: grid_units(180);
  margin-top: grid_units(4);
  margin-bottom: grid_units(4);
  background-color: inherit;

  &::before {
    content: '';
    position: absolute;
    top: 0;
    bottom: 0;
    left: calc(50% - 1px);
    width: 2px;
    background-color: $burgundy;
  }
}

.step {
  display: flex;
  align-items: flex-start;
  align-self: center;
  justify-content: center;
  width: 100%;
  background-color: inherit;
}

.stepImage {
  position: absolute;
  left: 50%;
  width: grid_units(24);
  height: grid_units(24);
  margin-left: grid_units(-12);
  border: none;
  border-radius: 50%;
  box-shadow: 0 0 20px -10px $rich-black;
}

.stepText {
  z-index: 1;
  max-width: grid_units(72);
  margin-bottom: grid_units(6);
  padding-top: grid_units(2);
  padding-bottom: grid_units(2);
  background-color: inherit;
}

.right {
  margin-top: grid_units(30);

  & > h4,
  & > p {
    background-color: inherit;
  }

  @media screen and (min-width: 940px) {
    min-height: grid_units(45);
    margin-top: 0;
    margin-left: grid_units(108);
    text-align: left;
  }
}

.left {
  margin-top: grid_units(28);

  & > h4,
  & > p {
    background-color: inherit;
  }

  @media screen and (min-width: 940px) {
    min-height: grid_units(45);
    margin-top: 0;
    margin-right: grid_units(108);
    text-align: right;
  }
}
