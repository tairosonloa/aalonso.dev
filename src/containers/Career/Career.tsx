import loadable from '@loadable/component'
import React, { FC } from 'react'
import { SectionContainer } from '../../components/DataDisplays/SectionContainer/SectionContainer'
import { SectionHeader } from '../../components/DataDisplays/SectionHeader/SectionHeader'
import styles from './Career.module.scss'

const CareerStep = loadable(() => import('../../components/DataDisplays/CareerStep/CareerStep'))

const career = [
  {
    imageUrl: '/images/acidtango.jpeg',
    date: '2019 - today',
    title: 'Full-Stack Developer',
    description:
      'Joined <a href="https://acidtango.com/" target="_blank" rel="noopener noreferrer">Acid Tango</a> to work as a full-stack Developer with Node.js and React. My stack is mainly TypeScript, using Next.js for the frontend and NestJS for the backend.',
  },
  {
    imageUrl: '/images/uc3m.png',
    date: '2019 - 2022 (expected)',
    title: "Master's Degree in Computer Science and Engineering",
    description:
      "Part-time studing a Master's Degree in Computer Science and Engineering at Universidad Carlos III de Madrid, while working full-time.",
  },
  {
    imageUrl: '/images/uc3m.png',
    date: '2017 - 2019',
    title: 'DevOps Developer Intern',
    description:
      'Working as intern in the Computer Science and Engineering Department at my university. DevOps tasks, including full-stack development and setting infrastructure.',
  },
  {
    imageUrl: '/images/techfest.png',
    date: '2016 - 2020',
    title: 'T3chfest Collaborator and Volunteer',
    description:
      'Helping making <a href="https://t3chfest.es/" target="_blank" rel="noopener noreferrer">T3chFest</a> possible for four years. Working near the organizers and friends to position T3chFest as a reference IT event.',
  },
  {
    imageUrl: '/images/uc3m.png',
    date: '2015 - 2019',
    title: "Bachelor's Degree in Computer Science and Engineering",
    description:
      "Got my Bachelor's Degree in Computer Science and Engineering at Universidad Carlos III de Madrid with honors. Specialized in artificial intelligence and data analysis.",
  },
]

export const Career: FC = () => {
  return (
    <SectionContainer sectionId="career" darkBackground>
      <SectionHeader title="Career" subtitle="My journey so far." />
      <ul className={styles.career}>
        {career.map((step, index) => (
          <li className={styles.step} key={step.title}>
            <img src={step.imageUrl} alt="" className={styles.stepImage} />
            <CareerStep
              date={step.date}
              title={step.title}
              description={step.description}
              stepIsOnTheLeft={index % 2 === 0}
            />
          </li>
        ))}
      </ul>
    </SectionContainer>
  )
}
