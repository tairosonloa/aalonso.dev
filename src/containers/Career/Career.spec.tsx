import { render } from '@testing-library/react'
import * as React from 'react'
import { Career } from './Career'

describe('<Career/>', () => {
  it('should display the default message', () => {
    const title = 'Career'
    const subtitle = 'My journey so far.'

    const renderResult = render(<Career />)

    expect(renderResult.getByText(title)).toBeTruthy()
    expect(renderResult.getByText(subtitle)).toBeTruthy()
  })

  it('should match the snapshot', () => {
    const renderResult = render(<Career />)
    expect(renderResult).toMatchSnapshot()
  })
})
