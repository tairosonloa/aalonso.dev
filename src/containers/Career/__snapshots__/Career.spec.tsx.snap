// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Career/> should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container dark"
      >
        <div
          class="container"
        >
          <h2
            class="common title"
            data-testid="section-title"
          >
            Career
          </h2>
          <h3
            class="common titleSmall"
            data-testid="section-subtitle"
          >
            My journey so far.
          </h3>
        </div>
        <ul
          class="career"
        >
          <li
            class="step"
          >
            <img
              alt=""
              class="stepImage"
              src="/images/acidtango.jpeg"
            />
            <div
              class="stepText left"
            >
              <h4
                class="common paragraphLead"
              >
                2019 - today
              </h4>
              <h4
                class="common paragraphLead"
              >
                Full Stack Developer
              </h4>
              <p
                class="common paragraph"
              >
                Joined 
                <a
                  href="https://acidtango.com/"
                  target="_blank"
                >
                  Acid Tango
                </a>
                 to work as a Full Stack Developer with NodeJs. My stack is mainly TypeScript, using React for the frontend and NestJS for the backend.
              </p>
            </div>
          </li>
          <li
            class="step"
          >
            <img
              alt=""
              class="stepImage"
              src="/images/uc3m.png"
            />
            <div
              class="stepText right"
            >
              <h4
                class="common paragraphLead"
              >
                2019 - 2022 (expected)
              </h4>
              <h4
                class="common paragraphLead"
              >
                Master's Degree in Computer Science and Engineering
              </h4>
              <p
                class="common paragraph"
              >
                Currently half-time studing a Master's Degree in Computer Science and Engineering at Universidad Carlos III de Madrid.
              </p>
            </div>
          </li>
          <li
            class="step"
          >
            <img
              alt=""
              class="stepImage"
              src="/images/uc3m.png"
            />
            <div
              class="stepText left"
            >
              <h4
                class="common paragraphLead"
              >
                2017 - 2019
              </h4>
              <h4
                class="common paragraphLead"
              >
                DevOps Developer Intern
              </h4>
              <p
                class="common paragraph"
              >
                Working as intern in the Computer Science and Engineering Department at my university. DevOps tasks, including full-stack development and setting infrastructure.
              </p>
            </div>
          </li>
          <li
            class="step"
          >
            <img
              alt=""
              class="stepImage"
              src="/images/techfest.png"
            />
            <div
              class="stepText right"
            >
              <h4
                class="common paragraphLead"
              >
                2016 - 2020
              </h4>
              <h4
                class="common paragraphLead"
              >
                T3chfest Collaborator and Volunteer
              </h4>
              <p
                class="common paragraph"
              >
                Helping making 
                <a
                  href="https://t3chfest.es/"
                  target="_blank"
                >
                  T3chFest
                </a>
                 possible for four years. Working near the organizers and friends to position T3chFest as a reference IT event.
              </p>
            </div>
          </li>
          <li
            class="step"
          >
            <img
              alt=""
              class="stepImage"
              src="/images/uc3m.png"
            />
            <div
              class="stepText left"
            >
              <h4
                class="common paragraphLead"
              >
                2015 - 2019
              </h4>
              <h4
                class="common paragraphLead"
              >
                Bachelor's Degree in Computer Science and Engineering
              </h4>
              <p
                class="common paragraph"
              >
                Got my Bachelor's Degree in Computer Science and Engineering at Universidad Carlos III de Madrid with honors. Specialized in artificial intelligence and data analysis.
              </p>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container dark"
    >
      <div
        class="container"
      >
        <h2
          class="common title"
          data-testid="section-title"
        >
          Career
        </h2>
        <h3
          class="common titleSmall"
          data-testid="section-subtitle"
        >
          My journey so far.
        </h3>
      </div>
      <ul
        class="career"
      >
        <li
          class="step"
        >
          <img
            alt=""
            class="stepImage"
            src="/images/acidtango.jpeg"
          />
          <div
            class="stepText left"
          >
            <h4
              class="common paragraphLead"
            >
              2019 - today
            </h4>
            <h4
              class="common paragraphLead"
            >
              Full Stack Developer
            </h4>
            <p
              class="common paragraph"
            >
              Joined 
              <a
                href="https://acidtango.com/"
                target="_blank"
              >
                Acid Tango
              </a>
               to work as a Full Stack Developer with NodeJs. My stack is mainly TypeScript, using React for the frontend and NestJS for the backend.
            </p>
          </div>
        </li>
        <li
          class="step"
        >
          <img
            alt=""
            class="stepImage"
            src="/images/uc3m.png"
          />
          <div
            class="stepText right"
          >
            <h4
              class="common paragraphLead"
            >
              2019 - 2022 (expected)
            </h4>
            <h4
              class="common paragraphLead"
            >
              Master's Degree in Computer Science and Engineering
            </h4>
            <p
              class="common paragraph"
            >
              Currently half-time studing a Master's Degree in Computer Science and Engineering at Universidad Carlos III de Madrid.
            </p>
          </div>
        </li>
        <li
          class="step"
        >
          <img
            alt=""
            class="stepImage"
            src="/images/uc3m.png"
          />
          <div
            class="stepText left"
          >
            <h4
              class="common paragraphLead"
            >
              2017 - 2019
            </h4>
            <h4
              class="common paragraphLead"
            >
              DevOps Developer Intern
            </h4>
            <p
              class="common paragraph"
            >
              Working as intern in the Computer Science and Engineering Department at my university. DevOps tasks, including full-stack development and setting infrastructure.
            </p>
          </div>
        </li>
        <li
          class="step"
        >
          <img
            alt=""
            class="stepImage"
            src="/images/techfest.png"
          />
          <div
            class="stepText right"
          >
            <h4
              class="common paragraphLead"
            >
              2016 - 2020
            </h4>
            <h4
              class="common paragraphLead"
            >
              T3chfest Collaborator and Volunteer
            </h4>
            <p
              class="common paragraph"
            >
              Helping making 
              <a
                href="https://t3chfest.es/"
                target="_blank"
              >
                T3chFest
              </a>
               possible for four years. Working near the organizers and friends to position T3chFest as a reference IT event.
            </p>
          </div>
        </li>
        <li
          class="step"
        >
          <img
            alt=""
            class="stepImage"
            src="/images/uc3m.png"
          />
          <div
            class="stepText left"
          >
            <h4
              class="common paragraphLead"
            >
              2015 - 2019
            </h4>
            <h4
              class="common paragraphLead"
            >
              Bachelor's Degree in Computer Science and Engineering
            </h4>
            <p
              class="common paragraph"
            >
              Got my Bachelor's Degree in Computer Science and Engineering at Universidad Carlos III de Madrid with honors. Specialized in artificial intelligence and data analysis.
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
